src/librbd/internal.cc

```diff
@@ -438,6 +439,28 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force,
     *opts = static_cast<rbd_image_options_t>(opts_);
   }
 
+  void image_options_copy(rbd_image_options_t* opts,
+			  ImageOptions &orig)
+  {
+    image_options_ref* opts_ = new image_options_ref(new image_options_t());
+
+    *opts = static_cast<rbd_image_options_t>(opts_);
+
+    for (auto &i : IMAGE_OPTIONS_TYPE_MAPPING) {
+      if (i.second == STR) {
+	std::string val;
+	if (orig.get(i.first, &val) == 0) {
+	  image_options_set(*opts, i.first, val);
+	}
+      } else if (i.second == UINT64) {
+	uint64_t val;
+	if (orig.get(i.first, &val) == 0) {
+	  image_options_set(*opts, i.first, val);
+	}
+      }
+    }
+  }
+
```

Comment:
So we have to iterate the image options and use the accessors (get() and set()) to do a full copy.